src/Features/LanguageServer/Protocol/CustomProtocol/FindUsagesLSPContext.cs

```diff
@@ -0,0 +1,220 @@
+            {
+                if (_resultsChunk.IsEmpty())
+                {
+                    return;
+                }
+
+                results = _resultsChunk.ToArray();
+                _resultsChunk.Clear();
+            }
+
+            _progress.Report(results);
+        }
+
+        public async override Task OnDefinitionFoundAsync(DefinitionItem definition)
+        {
+            using (await _semaphore.DisposableWaitAsync(CancellationToken).ConfigureAwait(false))
+            {
+                if (!_definitionToId.ContainsKey(definition))
+                {
+                    // Assinging a new id to the definition
+                    _definitionToId.Add(definition, _id);
+
+                    // VSReferenceItem currently doesn't support the ClassifiedTextElement type for DefinitionText,
+                    // so for now we just pass in a string.
+                    // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1100009
+                    var classifiedText = definition.GetClassifiedText();
+                    var definitionText = "";
+                    foreach (var text in classifiedText.Runs)
+                    {
+                        definitionText += text.Text;
+                    }
+
+                    // Creating a new VSReferenceItem for the definition
+                    var definitionItem = await GenerateVSReferenceItemAsync(
+                        _id, definitionId: _id, _document, _position, definition.SourceSpans.FirstOrDefault(),
+                        definition.DisplayableProperties, _metadataAsSourceFileService, definitionText,
+                        symbolUsageInfo: null, CancellationToken).ConfigureAwait(false);
+
+                    if (definitionItem.Location != null)
+                    {
+                        AddAndReportResultsIfAtMax(definitionItem);
+                        _id++;
+                    }
+                }
+            }
+        }
+
+        public async override Task OnReferenceFoundAsync(SourceReferenceItem reference)
+        {
+            using (await _semaphore.DisposableWaitAsync(CancellationToken).ConfigureAwait(false))
+            {
+                if (_definitionToId.TryGetValue(reference.Definition, out var definitionId))
+                {
+                    // Creating a new VSReferenceItem for the reference
+                    var referenceItem = await GenerateVSReferenceItemAsync(
+                        _id, definitionId, _document, _position, reference.SourceSpan,
+                        reference.AdditionalProperties, _metadataAsSourceFileService, definitionText: null,
+                        reference.SymbolUsageInfo, CancellationToken).ConfigureAwait(false);
+
+                    if (referenceItem.Location != null)
+                    {
+                        AddAndReportResultsIfAtMax(referenceItem);
+                        _id++;
```

Comment:
same comment as above.  this should happen almost immediately.  it's too easy to forget to do otherwise.